src/key_info_managers/mod.rs

```diff
@@ -98,7 +108,7 @@ pub trait ManageKeyInfo {
     /// # Errors
     ///
     /// Returns an error as a String if there was a problem accessing the Key Info Manager.
-    fn get_all(&self, provider_id: ProviderID) -> Result<Vec<&KeyTriple>, String>;
+    fn get_all(&self, provider_id: ProviderID) -> Result<Vec<(KeyTriple, KeyInfo)>, String>;
```

Comment:
What is the context of that change?

The reason we used a reference here was to mimick the different methods on the `HashMap` and here particularly [`get`](https://doc.rust-lang.org/beta/std/collections/struct.HashMap.html#method.get) which returns a reference.
I would be happy to add another method `to_mappings_vec` which has this signature or modify this one but using references instead. Like returning a `Vec<(&KeyTriple, Option<&KeyInfo>)>` for example?

Reply:
The context was - when initialising the PKCS11 provider with its new cache, we needed both the values and the key information. But if I remove the cache 

> The reason we used a reference here was to mimick the different methods on the HashMap and here particularly get which returns a reference.

That's quite forced, though, because `HashMap` does not have a `get_all` (it's called `keys` or `values`) and does not return a `Vec` of references (it returns an iterator). So the only thing common was the fact that it was returning references. I thought `get_all` as a name makes more sense with returning everything, but yeah, since the cache is gone we don't need it.